src/idl/src/annotation.c

```diff
@@ -129,6 +129,41 @@ annotate_autoid(
   return IDL_RETCODE_OK;
 }
 
+static idl_retcode_t
+annotate_optional(
+  idl_pstate_t *pstate,
+  idl_annotation_appl_t *annotation_appl,
+  idl_node_t *node)
+{
+  const idl_const_expr_t *const_expr;
+  bool value = true;
+
+  assert(annotation_appl);
+
+  if (!idl_is_member(node)) {
+    idl_error(pstate, idl_location(annotation_appl),
+      "@optional can only be assigned to member nodes");
+    return IDL_RETCODE_SEMANTIC_ERROR;
+  }
+
+  if (annotation_appl->parameters) {
+    const_expr = annotation_appl->parameters->const_expr;
+    assert(const_expr);
+    if (idl_mask(const_expr) != (IDL_BOOL | IDL_LITERAL)) {
+      idl_error(pstate, idl_location(annotation_appl),
+        "@optional cannot have non-boolean values assigned");
+      return IDL_RETCODE_SEMANTIC_ERROR;
+    } else {
+      value = ((const idl_literal_t*)const_expr)->value.bln;
+    }
```

Comment:
This is probably checked by the parser already. Like in one of my earlier comments, let's add a couple of small tests and fix the parser if it doesn't behave correctly already.

Reply:
You're correct that this should not really be checked here.
I will check whether the parser catches this, or if this requires a fix someplace else

Reply:
This is correctly being checked by the parser at a previous step, and therefore I will replace this check with a simpler assert